src/workerd/io/io-context.c++

```diff
@@ -1114,16 +1115,40 @@ void IoContext::runImpl(Runnable& runnable,
           // Ensure we don't pump the message loop in this case
           gotTermination = true;
         }
-      } while ([&]() {
-        if (gotTermination) {
-          // TerminateExecution() doesn't drain any tasks from platform's task queue
-          // so we want to avoid pumping the message loop in this case
-          return false;
-        } else {
-          SuppressIoContextScope noIoCtxt;
-          return js.pumpMsgLoop();
+      }
+
+      // Run FinalizationRegistry cleanup tasks without an IoContext
+      {
+        SuppressIoContextScope noIoCtxt;
+        while (!gotTermination && js.pumpMsgLoop()) {
+          // Check if FinalizationRegistry cleanup callbacks have not breached our limits
+          try {
+            limitEnforcer->requireLimitsNotExceeded();
```

Comment:
Use `limitEnforcer->getLimitsExceeded()` rather than throwing an exception and catching it here.